src/contracts/atlas/Atlas.sol

```diff
@@ -69,7 +69,7 @@ contract Atlas is Escrow, Factory {
         // Initialize the lock
         _setAtlasLock(executionEnvironment, gasMarker, userOp.value);
 
-        try this.execute(dConfig, userOp, solverOps, executionEnvironment, msg.sender, userOpHash) returns (
+        try this.execute(dConfig, userOp, solverOps, executionEnvironment, msg.sender, dAppOp.userOpHash) returns (
```

Comment:
This is a nice optimization. Maybe worth just adding a comment explaining that the dAppOp and solverOps would have the same version of userOpHash that was propagated - either TRUSTED or DEFAULT, and if we reach this part of the code then we know `dAppOp.userOpHash` is correct because it passed verification in `validateCalls()`

Reply:
f535659